coincontrol/main/views.py

```diff
@@ -3,6 +3,34 @@
 @main.route('/dashboard', methods=["GET"])
 def dashboard():
     pass
+@main.route('/incomes', methods=["POST", "GET"])
```

Comment:
You only defined one route for incomes which only retrieve the list of a particular users income and also let the user add an income . soo this doesnt track the income management routes section.

- Firstly you will need to define other routes for income which retrieves a specific income with a given ID.
- Secondly a route that updates the users income by using a specific ID.
- Thirdly a route that deletes a user income source with the given ID